crates/next-core/src/next_config.rs

```diff
@@ -0,0 +1,156 @@
+use std::collections::HashMap;
+
+use anyhow::Result;
+use serde::{Deserialize, Serialize};
+use turbo_tasks::{trace::TraceRawVcs, Value};
+use turbo_tasks_fs::FileSystemPathVc;
+use turbopack_core::{
+    chunk::ChunkingContextVc, context::AssetContextVc, virtual_asset::VirtualAssetVc,
+};
+use turbopack_ecmascript::{
+    EcmascriptInputTransform, EcmascriptInputTransformsVc, EcmascriptModuleAssetType,
+    EcmascriptModuleAssetVc,
+};
+use turbopack_node::read_config::{JavaScriptConfig, JavaScriptConfigVc, JavaScriptValue};
+
+use crate::embed_js::next_js_file;
+
+#[turbo_tasks::value(transparent)]
+pub struct NextConfigValue(NextConfig);
```

Comment:
Why is this a separate struct?

Reply:
I remember use `#[turbo_tasks::value]` directly will implement different`Serialize/Deserialize` trait for the `NextConfig`

Reply:
You can use `#[turbo_tasks::value(serialization = "custom")]` to manually implement serialization/deserialization.